### PR TITLE
Enhanced the UI part of Exercises by creating GymExercise and GymExer…

### LINKED ISSUE #40: Title: Enhance Body Parts Categories UI with GIFs
#40
Currently, the UI for the body parts categories displays static images, which could be further improved to create a more engaging user experience. I suggest replacing these static images with GIFs that demonstrate exercises for each body part. Not only will this make the interface more visually appealing, but it will also provide users with dynamic content that showcases exercises in action, ultimately enhancing their overall experience.

Proposed Changes:
GIF Integration: Replace static images with relevant GIFs for each body part category. I have GIFs available in the .json file that can be utilized for this purpose.

User Experience: This change will not only attract users' attention but will also help them visualize the exercises better, potentially increasing user engagement and satisfaction.

I want to enhance this UI part.
Screenshot (121)

I want to contribute this issue under hactoberfest and GSOC-EXT

I am willing to work on this.Kindly assign it to me.

### PR DESCRIPTION
**Pull request by rajlakshmi18704**
Enhanced the UI of Exercises Part and added the json file also as mentioned.

#40: Enhance Body Parts Categories UI with GIFs 

Title :Enhance Body Parts Categories UI with GIFs 

Issue No. :#40

Code Stack : 

Close #40

# Video (mandatory)

https://github.com/user-attachments/assets/05473d72-7800-4c55-baa2-481d00c872d1

# Checklist:

- [X ] I have mentioned the issue number in my Pull Request.
- [X ] I have commented my code, particularly in hard-to-understand areas
- [ X] I have created a helpful and easy to understand `README.md` if its a new page/tech stack
- [X ] I have gone through the  `contributing.md` file before contributing

**Additional context (Mandatory )**

***Are you contributing under any Open-source programme?***

- [ X] I'm a GSSOC-EXT contributor
- [ X] I'm a HACKTOBERFEST contributor

### CODE REVIEW

**Comment by Sami3160** [comment]
> Enhanced the UI of Exercises Part and added the json file also as mentioned.
> 
> #40: Enhance Body Parts Categories UI with GIFs
> 
> Title :Enhance Body Parts Categories UI with GIFs
> 
> Issue No. :#40
> 
> Code Stack :
> 
> Close #40
> # Video (mandatory)
> FitFlex-UI.mp4
> # Checklist:
> 
>     * [X ] I have mentioned the issue number in my Pull Request.
> 
>     * [X ] I have commented my code, particularly in hard-to-understand areas
> 
>     * [ X] I have created a helpful and easy to understand `README.md` if its a new page/tech stack
> 
>     * [X ] I have gone through the  `contributing.md` file before contributing
> 
> 
> **Additional context (Mandatory )**
> 
> _**Are you contributing under any Open-source programme?**_
> 
>     * [ X] I'm a GSSOC-EXT contributor
> 
>     * [ X] I'm a HACKTOBERFEST contributor

hey @rajlakshmi18704  nice work, but the cards on that pages are supposed to represent specific body parts and muscles not exercise name or type, after clicking on cards those cards, actual exercises respective to muscles on cards will start, 
that module is yet to be completed, 


and you have added 'see details' which is not expected, as that card will lead to exercise plan itself, it is not their to show details

basically we are not supposed to show exercise names, instead we need to show only targeted muscles, and when user clicks on it, some list of days will appear with the list of exercises, 

so on that other module where we will show list of exercise to do, we can use your gif thing. 

check issue #43 to understand more,
try using gif of muscles only,  and also try other card design that will match website theme, and add progressbar showing how much exercise user have completed and either add start button to start exercise or something else as you wish 


@rajlakshmi18704  really appreciate your hard work on this issue,

**Review by Sami3160 — CHANGES_REQUESTED** [review]
hey @rajlakshmi18704  nice work, but the cards on that pages are supposed to represent specific body parts and muscles not exercise name or type, after clicking on cards those cards, actual exercises respective to muscles on cards will start,
that module is yet to be completed,

and you have added 'see details' which is not expected, as that card will lead to exercise plan itself, it is not their to show details

basically we are not supposed to show exercise names, instead we need to show only targeted muscles, and when user clicks on it, some list of days will appear with the list of exercises,

so on that other module where we will show list of exercise to do, we can use your gif thing.

check issue https://github.com/Open-Code-Crafters/FitFlex/issues/43 to understand more,
try using gif of muscles only, and also try other card design that will match website theme, and add progressbar showing how much exercise user have completed and either add start button to start exercise or something else as you wish

@rajlakshmi18704  really appreciate your hard work on this issue,

**Comment by jeevan10017** [comment]
@rajlakshmi18704 updates?

**Comment by jeevan10017** [comment]
@rajlakshmi18704 updates?

**Comment by Sami3160** [comment]
> Enhanced the UI of Exercises Part and added the json file also as mentioned.
> 
> #40: Enhance Body Parts Categories UI with GIFs
> 
> Title :Enhance Body Parts Categories UI with GIFs
> 
> Issue No. :#40
> 
> Code Stack :
> 
> Close #40
> # Video (mandatory)
> FitFlex-UI.mp4
> # Checklist:
> 
>     * [X ] I have mentioned the issue number in my Pull Request.
> 
>     * [X ] I have commented my code, particularly in hard-to-understand areas
> 
>     * [ X] I have created a helpful and easy to understand `README.md` if its a new page/tech stack
> 
>     * [X ] I have gone through the  `contributing.md` file before contributing
> 
> 
> **Additional context (Mandatory )**
> 
> _**Are you contributing under any Open-source programme?**_
> 
>     * [ X] I'm a GSSOC-EXT contributor
> 
>     * [ X] I'm a HACKTOBERFEST contributor

@rajlakshmi18704 do you got any updates? or should i close this pr?

**Comment by rajlakshmi18704** [comment]
I want to ask that what change do I need to display is it like that I have to only put gifs in place of images ,rest details should be same?
I want to work on this issue this much and in issue no. 40 I can add more functionality.Is it ok or you want something more changes ?

Or I have the other idea also where I can create a seperate page for  exercises in which I will be displaying exercises  for different body parts  and equipments required. Do you want that?

**Comment by Sami3160** [comment]
> I want to ask that what change do I need to display is it like that I have to only put gifs in place of images ,rest details should be same? I want to work on this issue this much and in issue no. 40 I can add more functionality.Is it ok or you want something more changes ?
> 
> Or I have the other idea also where I can create a seperate page for exercises in which I will be displaying exercises for different body parts and equipments required. Do you want that?

look, this module is simple, just add have some broad(high level) categories of exercises and then with carousel like components show detailed(low level) exercises types, okay? 

on clicking on that exercise sub type we will be showing list of actual exercises(no types) which user will perform,
their you can implement these gif to show how exercise is done, 

for now while showing these gif for category and again use them again while performing exercises will not be good..i suggest instead of use them for showing categories you do the next module, which is creating interface for doing actual exercises where you can use these gif, their it can be very helpful, check issue #43 , that what you need to do next

**Comment by GAVINESHWAR** [comment]
@rajlakshmi18704, This PR is having a conflict please resolve it.

**Comment by GAVINESHWAR** [comment]
@rajlakshmi18704, This PR is having a conflict please resolve it. Otherwise i will close the PR.

**Comment by Sami3160** [comment]
> I want to ask that what change do I need to display is it like that I have to only put gifs in place of images ,rest details should be same? I want to work on this issue this much and in issue no. 40 I can add more functionality.Is it ok or you want something more changes ?
> 
> Or I have the other idea also where I can create a seperate page for exercises in which I will be displaying exercises for different body parts and equipments required. Do you want that?

@rajlakshmi18704 are you still working on this pr?
